### PR TITLE
Restrict functions arity

### PR DESCRIPTION
**Pull request by erickpintor**
Closes: https://github.com/faunadb/sales-engineering/issues/379

### CODE REVIEW

File: src/query.js [hunk 1]
```diff
@@ -744,11 +801,27 @@ function Or() {
...
 
 // Helpers
 
+/**
+ * @ignore
+ */
+function arity(min, max, message, args) {
+  if ((min !== null && args.length < min) || (max !== null && args.length > max)) {
+    throw new errors.InvalidValue(
+      'Function require ' + message + ' arguments but ' + args.length + ' were given.');
```
Comment (eaceaser) [inline src/query.js]: typo: require => requires

File: src/query.js [hunk 2]
```diff
@@ -744,11 +801,27 @@ function Or() {
...
 }
 
 // Helpers
 
+/**
+ * @ignore
+ */
+function arity(min, max, message, args) {
+  if ((min !== null && args.length < min) || (max !== null && args.length > max)) {
+    throw new errors.InvalidValue(
```
Comment (eaceaser) [inline src/query.js]: On second thought, should we just create another exception class and expose the arity information there, so we don't have to do regex string matching on the message to determine if the correct error was thrown?

Comment (jchris) [inline src/query.js]: I implemented that in dafd1e8 but I'm not attached to it either way.

**Comment by erickpintor** [comment]
@ed can you do another pass?

**Comment by erickpintor** [comment]
Is there anything else missing for this PR?

**Comment by jchris** [comment]
The only thing I can think of is that the tests don't actually use the InvalidArity features, and the message composition for that error type is still implemented in query.js as a helper, instead of being extracted to the error class. It seemed there was some additional refactoring needed here that would be obvious to the first person who starts to modify the InvalidArity type, so maybe it's better to wait and do that later? I'm happy to write the code either way I guess it depends on the prevailing winds as far as late-binding in our client code.

**Comment by erickpintor** [comment]
I have moved the error message composition to the error class.

Regarding make usage of arity features on the tests, I'm a little hesitant to make arity function public. It doesn't seems to be useful as a public interface at the driver since they are safe guards due to the lack of a more strict language option.

**Comment by eaceaser** [comment]
lgtm

**Pull request merged**